### PR TITLE
15 create calib plugin

### PR DESCRIPTION
**Pull request by wnichols1**
open to comments; I'm not saying this is neat!

### CODE REVIEW

File: data/frameProcessor/include/SIMDMacros.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#define Cast4fto4i(x) _mm_castps_si128(x)
+#define Cast8fto8i(x) _mm256_castps_si256(x)
+#define Cast8ito8f(x) _mm256_castsi256_ps(x)
+// use _mm_permute_pd here
+#define SwapHalves4i(x) Cast4fto4i(_mm_permute_ps(Cast4ito4f(x),0x4e))
+// cvt_epu_ does zero extension!
+#define ExtendLoto4i(x) _mm_cvtepu16_epi32(x)
+// could use _mm_bsrli_si128 here byte shift right
+#define ExtendHito4i(x) _mm_cvtepu16_epi32(SwapHalves4i(x))
+#define Extend4ito8i(x) Set8i(ExtendLoto4i(x), ExtendHito4i(x))
```
Comment (wnichols1) [inline data/frameProcessor/include/SIMDMacros.h]: rename, note the avx2 macro

File: data/frameProcessor/include/SIMDMacros.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+// avx is missing integer operations on 8i, so we may need 4i registers from SSE.
+
+typedef __m256 SIMD8f;  // 8 floats
+typedef __m128i SIMD4i; // 4 integers
+typedef __m128i SIMD8i16; // 8x 16 bit integers
+typedef __m256i SIMD8i;
+
+
+#define SIMD_MEM_ALIGN 32
```
Comment (wnichols1) [inline data/frameProcessor/include/SIMDMacros.h:31]: use this in the simdtests.cpp

File: data/frameProcessor/include/SIMDMacros.h [hunk 3]
```diff
@@ -0,0 +1,131 @@
+
```
Comment (wnichols1) [inline data/frameProcessor/include/SIMDMacros.h:1]: add name and copyright notice

File: data/frameProcessor/src/CMakeLists.txt [hunk 4]
```diff
@@ -2,10 +2,29 @@
 set(CMAKE_INCLUDE_CURRENT_DIR on)
 ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
 
+set(HDF5_ROOT "/dls/detectors/LATRD/software/ext_src/hdf5/1-10-6/prefix/")
```
Comment (ajgdls) [inline data/frameProcessor/src/CMakeLists.txt]: This line won't work outside of Diamond.  Also it points to a LATRD (Tristan) detector location.

File: data/frameProcessor/src/CMakeLists.txt [hunk 5]
```diff
@@ -2,10 +2,29 @@
...
 add_library(PercivalProcessPlugin SHARED PercivalProcessPlugin.cpp)
+add_library(PercivalGenPlugin SHARED PercivalGenPlugin.cpp)
+add_library(PercivalCalibPlugin SHARED PercivalCalibPlugin.cpp)
+
+# warning this applies to the whole file - can we do better?
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx -fpic")
+add_library(PercivalCalib STATIC CalibratorSample.cpp CalibratorReset.cpp Calibrator.cpp FrameMem.cpp)
+target_include_directories(PercivalCalib PRIVATE "${HDF5_ROOT}/include")
+
+target_link_libraries(PercivalCalib PUBLIC /usr/lib64/libtbb.so ${HDF5_ROOT}/lib/libhdf5.so ${HDF5_ROOT}/lib/libhdf5_hl.so)
```
Comment (ajgdls) [inline data/frameProcessor/src/CMakeLists.txt]: Does it need that path or is /usr/lib64 not already in the search path?  What happens if tbb is installed in another location.

**Review by ajgdls — COMMENTED** [review]
I still need to build this module and check I can run it, so just added a few specific comments regarding the build.

File: data/frameProcessor/src/CMakeLists.txt (hunk 4 above)
Comment (wnichols1) [reply data/frameProcessor/src/CMakeLists.txt]: ok Ive moved this to the script that runs cmake

File: data/frameProcessor/src/CMakeLists.txt (hunk 5 above)
Comment (wnichols1) [reply data/frameProcessor/src/CMakeLists.txt]: I've taken that path out and just put libtbb.so
I hope that works!

**Review by ajgdls — APPROVED** [review]

**Pull request merged**